alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/notification/NotificationProcessor.java

```diff
@@ -112,6 +154,46 @@ public NotificationProcessor(ConfigurationAccessor configurationAccessor, List<P
         return List.of();
     }
 
+    private List<DistributionEvent> processNotifications(StatefulProvider statefulProvider, ProviderDistributionFilter distributionFilter, ConfigurationJobModel job, List<AlertNotificationModel> notifications) {
+        if (!job.isEnabled()) {
+            logger.debug("Skipping disabled distribution job: {}", job.getName());
+            return List.of();
+        }
+        if (notifications.isEmpty()) {
+            return List.of();
+        }
```

Comment:
The method containing these statements is private and only invoked in one place, we should be able to short-circuit much earlier on these checks.